com.microsoft.mrtk.input/Utilities/InputActionPoseSource.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    [Serializable]
+    public class InputActionPoseSource : IPoseSource
+    {
+        [SerializeField]
+        InputActionProperty trackingStateActionProperty;
+
+        [SerializeField]
+        InputActionProperty positionActionProperty;
+
+        [SerializeField]
+        InputActionProperty rotationActionProperty;
+
+        public bool TryGetPose(out Pose pose)
+        {
+            InputAction trackingStateAction = trackingStateActionProperty.action;
+            InputAction positionAction = positionActionProperty.action;
+            InputAction rotationAction = rotationActionProperty.action;
+
+            if (trackingStateAction != null && trackingStateAction.enabled
+                && positionAction != null && positionAction.enabled
+                && rotationAction != null && rotationAction.enabled)
```

Comment:
We should probably also check that the tracking state is actually tracked!

```suggestion
                && rotationAction != null && rotationAction.enabled
                && ((InputTrackingState)trackingStateAction.ReadValue<int>() & (InputTrackingState.Position | InputTrackingState.Rotation)) != 0)
)
```